engines/bops_api/app/services/bops_api/application/creation_service.rb

```diff
@@ -136,6 +136,45 @@ def process_ownership_certificate_details(planning_application)
           end
         end
       end
+
+      def possibly_immune?(planning_application)
+        # TODO make planning_application an instance variable to simply this method et seq.
+        planning_application.immune_proposal_details.count > 1
+      end
+
+      def process_immunity_details(planning_application)
+        ActiveRecord::Base.transaction do
+          immunity_detail = ImmunityDetail.new(planning_application: planning_application)
+          immunity_detail.end_date = planning_application
+            .find_proposal_detail("When were the works completed?")
```

Comment:
Should we be suggesting adding something to the schema for this?

Reply:
Yeah, I think it would be great if they would provide a slightly more processed form of these questions. I also think it's not ideal that we have to hunt through an array for an English-language string, etc.